pkg/subctl/cmd/gather/summary.go

```diff
@@ -0,0 +1,221 @@
+/*
+© 2021 Red Hat, Inc. and others
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package gather
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"html/template"
+	"io"
+	"os"
+	"path/filepath"
+
+	subctlversion "github.com/submariner-io/submariner-operator/pkg/version"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// Embed the file content as string.
+//go:embed layout.gohtml
+var layout string
+
+type version struct {
+	Subctl    string
+	Subm      string
+	K8sServer string
+}
+
+type clusterConfig struct {
+	CNIPlugin        string
+	CloudProvider    string
+	TotalNode        int
+	GatewayNode      map[string]types.UID
+	GWNodeNumber     int
+	MasterNode       map[string]types.UID
+	MasterNodeNumber int
+}
+
+type nodeConfig struct {
+	Name string
+	Info v1.NodeSystemInfo
+}
+
+type logInfo struct {
+	PodName      string
+	Namespace    string
+	RestartCount int32
+	PodState     v1.PodPhase
+	LogFileName  []string
+}
+
+type resourceInfo struct {
+	Name      string
+	Namespace string
+	Type      string
+	FileName  string
+}
+
+type data struct {
+	ClusterName   string
+	Versions      version
+	ClusterConfig clusterConfig
+	NodeConfig    []nodeConfig
+	PodLogs       []logInfo
+	ResourceInfo  []resourceInfo
+}
+
+func ClusterInformation(info Info) {
```

Comment:
Can we think of a better name? 

WriteClusterInfoReport ?

Reply:
This will be called with gather. `gather.WriteClusterInfoReport` does not look good. `gather.ClusterInformation` or may be `gather.ClusterSummary` is more meaningful

Reply:
+1 for `ClusterSummary`